### ISSUE TITLE
Cannot control several standalone apps

### ISSUE THREAD

**Issue by efylan** [rg35xx-cfw/Koriki#46]
### What happened?

For some reason,  I cannot do anything within several apps, not all. I already tried the bootfix thing, too.

#### Flycast standalone app
When I boot a game from the list to flycast standalone, the game does boot, but I cannot do anything with the dpad or menu keys or triggers, I'm guessing it's some kind of configuration error. But I cannot even navigate to the settings to change it.

#### Commander
Cannot do a single thing either, controls are unresponsive.

#### ScummVM

Can only  open the keyboard with L2 (not use it) and exit with Y

#### Wifi app

Controls unresponsive

Tested with a fresh install.

Edit:

### I think I managed to trace the problem, it's the line for setting the SDL_GAMECONTROLLERCONFIG environment variable, which has a typo.

export SDL_GAMECONTROLLERCONFIG=$(grep "RG35XX" "${HOME}/.config/gamecontrolerdb.txt")

The file name should be gamecontrollerdb.txt. I couldn't fin the location of this on the repo, if you can point me to it I can change it.

Post installation, the shell script files that need this to run live within /system/.simplemenu/apps/
So if someone has this problem, check those shell launchers and correct the typo. Just confirmed that correcting this path makes it work. I wonder why some other apps like drastic have this line yet have the controllers perfectly mapped.

**Comment by acmeplus** [comment]
Basically you need a game gamecontrollerdb.txt file in that location that contains at least this line:
```
19000000010000000100000000010000,RG35XX Gamepad,a:b0,b:b1,x:b2,y:b3,leftshoulder:b4,rightshoulder:b5,lefttrigger:b6,righttrigger:b7,guide:b8,start:b9,back:b10,dpup:b13,dpleft:b15,dpright:b16,dpdown:b14,volumedown:b11,volumeup:b12,platform:Linux,
```

The rest of the file can be the same as the standard file from the SDL_GameControllerDB repo: https://github.com/gabomdq/SDL_GameControllerDB/blob/master/gamecontrollerdb.txt

**Comment by acmeplus** [comment]
Note that I've not been able to get standalone ScummVM to work. All key mappings that I've tried have not resulted in any working configuration.

**Comment by efylan** [comment]
I just mapped the controls for the pcsx standalone emulator. If anyone is interested on mapping those for 0.9.1 you can replace pcsx.cfg with this file in both /usr/share/.pcsx/ and system/.pcsx/

pcsx standalone config file here:
https://gist.github.com/efylan/8e36e6af73e1608ebd4d439ee521f5cf

**Comment by efylan** [comment]
> Note that I've not been able to get standalone ScummVM to work. All key mappings that I've tried have not resulted in any working configuration.

Can I see your scummvm.ini attempts? Or are you trying it on build?
Is this useful? https://retropie.org.uk/forum/topic/16407/howto-multiple-controller-configs-in-scummvm-virtual-kbd

**Comment by acmeplus** [comment]
> I just mapped the controls for the pcsx standalone emulator. If anyone is interested on mapping those for 0.9.1 you can replace pcsx.cfg with this file in both /usr/share/.pcsx/ and system/.pcsx/
> 
> pcsx standalone config file here: https://gist.github.com/efylan/8e36e6af73e1608ebd4d439ee521f5cf

One of the things I need to figure out is how to patch pcsx to show its menu. Did you check if your mappings are different from the ones already included?

**Comment by acmeplus** [comment]
> > Note that I've not been able to get standalone ScummVM to work. All key mappings that I've tried have not resulted in any working configuration.
> 
> Can I see your scummvm.ini attempts? Or are you trying it on build? Is this useful? https://retropie.org.uk/forum/topic/16407/howto-multiple-controller-configs-in-scummvm-virtual-kbd

I don't have any recent files saved, that was weeks ago and I didn't save them, but in general all my attempts to define direct mappings for the vkb in scummvm.ini failed. That link may be a good path to just hardcode the rg35xx definitions in a patch. I'll look into that.

**Comment by papkocaro3** [comment]
I have  questions.

I put gamecontrollerdb.txt with the description of Mr.acmeplus 
in userdata/system/.config/ folder (Tentatively Also gamecontrolerdb.txt)

But the controller doesn't work.

Am I targeting it wrong?

Or, is it because it is a V6C board?


I will add, but for koriki 0.9 or 0.9.1
It's a problem.

Below 0.8, the Start button becomes A and some buttons do not work,
 but it works somehow

**Comment by k5jat** [comment]
Exactly where do I need to change the environment variable to the correct file spelling? It would be nice to get a quick update on how to fix this or just a fix patch/update would be good too.

**Comment by acmeplus** [comment]
> Exactly where do I need to change the environment variable to the correct file spelling? It would be nice to get a quick update on how to fix this or just a fix patch/update would be good too.

Yeah. At this point I'll publish a patch (basically the system folder) for users to test and provide feedback. There's still work to consolidate the kernel of the different motherboards but 99% of the issue folks have are related to configuration issues.

**Comment by k5jat** [comment]
> > Exactly where do I need to change the environment variable to the correct file spelling? It would be nice to get a quick update on how to fix this or just a fix patch/update would be good too.
> 
> Yeah. At this point I'll publish a patch (basically the system folder) for users to test and provide feedback. There's still work to consolidate the kernel of the different motherboards but 99% of the issue folks have are related to configuration issues.

Thank you! You're very much appreciated!

**Comment by acmeplus** [comment]
> > Yeah. At this point I'll publish a patch (basically the system folder) for users to test and provide feedback. There's still work to consolidate the kernel of the different motherboards but 99% of the issue folks have are related to configuration issues.
> 
> Thank you! You're very much appreciated!

Check the latest system update release, that's basically an update system folder:
https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v0.9.2-test

**Comment by k5jat** [comment]
I downloaded it and tried it. The Wifi app now takes up a quarter of the screen and I'm still not able to use the controls in it. Everything seems to be working fine for me.

**Comment by acmeplus** [comment]
Just an update to mention that I've managed to get scummvm standalone controls to work but still need to figure out how to properly add that support into the build.
